java-frontend/src/main/java/org/sonar/java/se/constraint/TypedConstraint.java

```diff
@@ -19,6 +19,33 @@
  */
 package org.sonar.java.se.constraint;
 
+import org.sonar.plugins.java.api.semantic.Type;
+
+import javax.annotation.Nullable;
+import java.util.Objects;
+
 public class TypedConstraint implements Constraint {
-  // Empty class for now, but should store the resolved type for instanceof operator.
+  @Nullable
+  public final Type type;
```

Comment:
I am wondering about not having this field nullable and rather rely on Symbols.unknownType